Tests/SonarQube.Common.UnitTests/SarifTests.cs

```diff
@@ -0,0 +1,293 @@
+          ""analysisTarget"": [
+            {
+              ""uri"": ""C:\\agent\\_work\\2\\s\\MyTestProj\\Program.cs"",
+}
+          ]
+        }
+      ],
+      ""shortMessage"": ""Test shortMessage. It features \""quoted text\""."",
+      ""properties"": {
+        ""severity"": ""Info"",
+        ""helpLink"": ""https://github.com/SonarSource/sonar-msbuild-runner"",
+      }
+    }
+  ]
+}", fixedSarif);
+        }
+
+        [TestMethod]
+        public void Utilities_SarifFixer_ShouldReplaceBackslashes()
+        {
+            // Arrange
+            string testSarif = @"{
+  ""version"": ""0.1"",
+  ""toolInfo"": {
+                ""toolName"": ""Microsoft (R) Visual C# Compiler"",
+    ""productVersion"": ""1.0.0"",
+    ""fileVersion"": ""1.0.0""
+  },
+  ""issues"": [
+    {
+      ""ruleId"": ""DD001"",
+      ""locations"": [
+        {
+          ""analysisTarget"": [
+            {
+              ""uri"": ""C:\agent\_work\2\s\MyTestProj\Program.cs"",
+}
+          ]
+        }
+      ],
+    }
+  ]
+}";
+
+            // Act
+            string fixedSarif;
+            bool changeApplied = SarifUtilities.FixImproperlyEscapedSarif(testSarif, out fixedSarif);
+
+            // Assert
+            Assert.IsTrue(changeApplied);
+            Assert.AreEqual(@"{
+  ""version"": ""0.1"",
+  ""toolInfo"": {
+                ""toolName"": ""Microsoft (R) Visual C# Compiler"",
+    ""productVersion"": ""1.0.0"",
+    ""fileVersion"": ""1.0.0""
+  },
+  ""issues"": [
+    {
+      ""ruleId"": ""DD001"",
+      ""locations"": [
+        {
+          ""analysisTarget"": [
+            {
+              ""uri"": ""C:\\agent\\_work\\2\\s\\MyTestProj\\Program.cs"",
+}
+          ]
+        }
+      ],
+    }
+  ]
+}", fixedSarif);
+        }
+
+        [TestMethod]
+        public void Utilities_SarifFixer_ShouldReplaceQuotes()
+        {
+            // Arrange
+            string testSarif = @"{
+  ""version"": ""0.1"",
+  ""toolInfo"": {
+                ""toolName"": ""Microsoft (R) Visual C# Compiler"",
+    ""productVersion"": ""1.0.0"",
+    ""fileVersion"": ""1.0.0""
+  },
+  ""issues"": [
+    {
+      ""ruleId"": ""DD001"",
+      ""shortMessage"": ""Test shortMessage. It features ""quoted text""."",
+      ""properties"": {
+        ""severity"": ""Info"",
+        ""helpLink"": ""https://github.com/SonarSource/sonar-msbuild-runner"",
+      }
+    }
+  ]
+}";
+
+            // Act
+            string fixedSarif;
+            bool changeApplied = SarifUtilities.FixImproperlyEscapedSarif(testSarif, out fixedSarif);
+
+            // Assert
+            Assert.IsTrue(changeApplied);
+            Assert.AreEqual(@"{
+  ""version"": ""0.1"",
+  ""toolInfo"": {
+                ""toolName"": ""Microsoft (R) Visual C# Compiler"",
+    ""productVersion"": ""1.0.0"",
+    ""fileVersion"": ""1.0.0""
+  },
+  ""issues"": [
+    {
+      ""ruleId"": ""DD001"",
+      ""shortMessage"": ""Test shortMessage. It features \""quoted text\""."",
```

Comment:
This doesn't check *title* or *fullMessage* fields are fixed.
What about values that contain both quotes and backslashes?